manifests/plugin/intel_pmu.pp

```diff
@@ -0,0 +1,15 @@
+# https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_intel_pmu
+class collectd::plugin::intel_pmu (
+  Enum['present', 'absent'] $ensure                       = 'present',
+  Optional[Boolean]         $report_hardware_cache_events = undef,
```

Comment:
What are the defaults from collected? Is it possible to default to True or False, instead of undef? This would be cleaner.

Reply:
According to the [documentation](https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_intel_pmu), there are no defaults, therefore the defaults are an implementation detail, which is why I did not set any defaults as well. However, by inspecting the source, as of the latest version the default is that no events are selected at all.

Therefore I suggest to leave the boolean variables as false by default, to make this clear, and to force the behaviour of not selecting any PMU events by default.

Regarding parameters `event_list` and `hardware_events`, I believe it would be best to leave them undefined by default. According to the docs these are used together, and to define `hardware_events` one should also use `event_list`. Therefore I think it is more coherent to have them both undefined by default rather than to have `event_list` undefined but `hardware_events` as an empty list. I think that if `hardware_events` will not be used, it should not be defined and it should not end up in the resulting `intel_pmu.conf`.

One improvement that could be made to the current logic is to check that if hardware_events is defined, event_list should also be defined. I can submit a new commit with this.